client/src/controllers/DropdownController.ts

```diff
@@ -162,50 +146,90 @@ export class DropdownController extends Controller<HTMLElement> {
       });
     }
 
+    const onCreate = (instance: Instance<Props>) => {
+      if (this.keepMountedValue) {
+        const { popper } = instance;
+        this.element.append(popper);
+        popper.hidden = true;
+      }
+    };
+
+    const onShow = (instance: Instance<Props>) => {
+      if (hoverTooltipInstance) {
+        hoverTooltipInstance.disable();
+      }
+      if (this.keepMountedValue) {
+        const { popper } = instance;
+        popper.hidden = false;
+      }
+    };
+
     const onShown = () => {
       this.dispatch('shown');
     };
 
     const onHide = () => {
       this.dispatch('hide');
+      if (hoverTooltipInstance) {
+        hoverTooltipInstance.enable();
+      }
+    };
+
+    const onHidden = (instance: Instance<Props>) => {
+      if (this.keepMountedValue) {
+        const { popper } = instance;
+        this.element.append(popper);
+        popper.hidden = true;
+      }
     };
 
     return {
       ...(this.hasContentTarget
         ? { content: this.contentTarget as Content }
         : {}),
-      ...themeOptions[this.themeValue],
+      ...this.themeOptions,
       trigger: 'click',
       interactive: true,
       ...(this.hasOffsetValue && { offset: this.offsetValue }),
-      hideOnClick: !this.useCustomHideOnClickAway,
       getReferenceClientRect: () => this.reference.getBoundingClientRect(),
       theme: this.themeValue,
-      plugins: this.plugins,
-      onShow() {
-        if (hoverTooltipInstance) {
-          hoverTooltipInstance.disable();
-        }
-      },
-      onShown() {
-        onShown();
-      },
-      onHide() {
-        if (hoverTooltipInstance) {
-          hoverTooltipInstance.enable();
-        }
-        onHide();
-      },
+      onCreate,
+      onShow,
+      onShown,
+      onHide,
+      onHidden,
     };
   }
 
-  get useCustomHideOnClickAway() {
-    // Tippy's default hideOnClick option for "hiding on click away" doesn't
-    // work well with our datetime libraries. Instead, we use a custom plugin
-    // to hide the tooltip when clicking outside the dropdown. It's unlikely
-    // we'll render a datetime picker for themes other than drilldown, so use
-    // the custom solution for this theme only.
-    return this.themeValue === 'drilldown';
+  get themeOptions() {
+    return (
+      {
```

Comment:
While moving things around. Maybe it make more sense to have the options as a class static property?

```ts
export class DropdownController extends Controller<HTMLElement> {
  static themeOptions = {
    'dropdown': {
      arrow: true,
      maxWidth: 350,
      placement: 'bottom',
    },
    'drilldown': {
      arrow: false,
      maxWidth: 'none',
      placement: 'bottom-end',
    },
    'dropdown-button': {
      arrow: false,
      maxWidth: 'none',
      placement: 'bottom-start',
    },
  };
// ...

```

Then you can use it like

```ts

  get themeOptions() {
    return (this.constructor as typeof DropdownController).themeOptions[
      this.themeValue
    ];
  }
```

Note that TS does not consider the constructor the same way as you would expect, hence the `as typeof DropdownController`.

See https://github.com/microsoft/TypeScript/issues/3841

Reply:
Unfortunately I don't think I can do this, since the options for the `drilldown` theme includes a plugin for the custom clickaway handler that requires access to `this` instance of the controller. If you have ideas on how to approach this (preferably without moving the theme-specific `plugins` to be outside of the `themeOptions`), please let me know!

Reply:
Ahh. I missed that plugins item. 

The plugins and options could be split but at this point it's just nitpicky code changes that really don't matter to much. 

If we ever need to access these raw (non-plugins) values outside this code we may need to refactor but that can be a job for another day. 


Thanks for replying.